### PR TITLE
Cards and Carousel fixes

### PR DESCRIPTION
**Pull request by danmarshall**
Merged WrappedActivities with ActivityView.
Fixed size of cards.
Added overflow measurement to carousel so that it respects the chat width.
Removed "table" css.
Add padding to messages to "autoscroll" mimicking vertical-align:bottom.

### CODE REVIEW

File: src/Chat.tsx [hunk 1]
```diff
@@ -168,6 +168,21 @@ export const sendFiles = (dispatch: Dispatch<HistoryAction>, files: FileList, fr
...
+export const measure = { 
+    innerHeight: (el: HTMLElement) : number => {
+        var s = window.getComputedStyle(el);
+        var paddingTop = parseInt(s.getPropertyValue('padding-top'));
+        var paddingBottom = parseInt(s.getPropertyValue('padding-bottom'));
+        return el.offsetHeight - paddingTop - paddingBottom;
+    },
+    outerWidth: (el: HTMLElement) : number => {
+        var s = window.getComputedStyle(el);
+        var marginLeft = parseInt(s.getPropertyValue('margin-left'));
```
Comment (danmarshall) [inline src/Chat.tsx]: I suppose I could write a helper function to parse a propertyValue

**Comment by danmarshall** [comment]
fixes #230, #133, #114

**Comment by danmarshall** [comment]
@billba can you take a look at this PR? I'd like to move to the next batch :)

File: botchat.css [hunk 2]
```diff
@@ -122,6 +116,11 @@
         position: relative;
     }
 
+    .wc-message-wrapper.carousel .wc-message {
```
Comment (danmarshall) [inline botchat.css]: this is correct - it is selecting a wc-message inside of a wrapper with the carousel layout

Comment (billba) [inline botchat.css]: but your code is generating a space between `.wc-message-wrapper` and `.carousel`

File: src/History.tsx [hunk 3]
```diff
@@ -119,6 +119,26 @@ export const History = connect(
...
 )(HistoryContainer)
 
+const getComputedStyleValues = (el: HTMLElement, stylePropertyNames: string[]) => {
+    const s = window.getComputedStyle(el);
+    const result: { [key: string]: number } = {};
+    stylePropertyNames.forEach(name => result[name] = parseInt(s.getPropertyValue(name)));
+    return result;
+}
+
+const measure = {
```
Comment (billba) [inline src/History.tsx]: Please just make these two separate functions. We don't use this convention anywhere else in this codebase.

Comment (danmarshall) [inline src/History.tsx]: ok, I was preparing for the possibility of the other combinatorials (outerHeight, innerWidth...)

Comment (danmarshall) [inline src/History.tsx]: taking a cue from Jquery

File: botchat.css (hunk 2 above)
Comment (danmarshall) [inline botchat.css]: DOM and CSS selectors are separate syntaxes. In DOM, className are indeed separated by a space character, but in CSS selectors they have no space character. Another example would be an ID - in the DOM it's an attribute e.g. id="foo" but in CSS it's #foo

Comment (danmarshall) [inline botchat.css]: (reprint- comment did't appear here)
DOM and CSS selectors are separate syntaxes. In DOM, className are indeed separated by a space character, but in CSS selectors they have no space character. Another example would be an ID - in the DOM it's an attribute e.g. id="foo" but in CSS it's #foo

**Pull request merged**